In [1]:
#import library
from libsbml import readSBML
from cobra import io
from cobra.core import Metabolite
from Functions import find_average

#import sbml file
model = io.sbml.create_cobra_model_from_sbml_file("/home/sanu/Documents/Scripts/git/plantcoremetabolism-model/PlantCoreMetabolism_v1_2_3.xml")

#change Biomass_tx to Protein_biomass
rxn = model.reactions.get_by_id("Biomass_tx")
rxn.id = "Protein_biomass_tx"
for met in ["K_b","Ca_b","Mg_b"]:
    met  = model.metabolites.get_by_id(met)
    coeff = rxn.metabolites.get(met)
    rxn.add_metabolites({met:-1*coeff})
met = Metabolite("Protein_b",name="Protein_biomass")
formula_dict = rxn.check_mass_balance()
met.formula = "".join([atom+str(formula_dict[atom]*-1) for atom in formula_dict.keys() if atom != "charge"])
met.charge = formula_dict["charge"]*-1
rxn.add_metabolites({met:1})

In [2]:
import pandas as pd
biomass = pd.DataFrame(data={"":["sSUCROSE_b","GLC_c","FRU_c","Starch_b","Cellulose_b","PALMITATE_c",
                                 "L_PHOSPHATIDATE_p","PHOSPHATIDYL_CHOLINE_r",
                                 "L_1_PHOSPHATIDYL_ETHANOLAMINE_r","DIACYLGLYCEROL_r","Protein_b","sMAL_b",
                                 "sCIT_b","sFUM_b","ARG_c","HIS_c","LYS_c","sASP_b",
                                 "sGLU_b","sSER_b","THR_c","ASN_c","sGLN_b","CYS_c",
                                 "GLY_c","PRO_c","sALA_b","VAL_c","ILE_c","LEU_c",
                                 "MET_c","PHE_c","TYR_c","TRP_c","sGABA_b"],
                             "leaf":[0.0]*35,"stem":[0.0]*35,"root":[0.0]*35,"seed":[0.0]*35,},dtype="float64")
biomass = biomass.set_index("")

In [3]:
proteinMW = model.metabolites.get_by_id("Protein_b").formula_weight

cobra/core/metabolite.py:103 UserWarning: 17.2020450391 is not an integer (in formula C17.2020450391H25.9492553613O5.7497158698N4.9536245179S0.1420323181)
cobra/core/metabolite.py:103 UserWarning: 25.9492553613 is not an integer (in formula C17.2020450391H25.9492553613O5.7497158698N4.9536245179S0.1420323181)
cobra/core/metabolite.py:103 UserWarning: 5.7497158698 is not an integer (in formula C17.2020450391H25.9492553613O5.7497158698N4.9536245179S0.1420323181)
cobra/core/metabolite.py:103 UserWarning: 4.9536245179 is not an integer (in formula C17.2020450391H25.9492553613O5.7497158698N4.9536245179S0.1420323181)
cobra/core/metabolite.py:103 UserWarning: 0.1420323181 is not an integer (in formula C17.2020450391H25.9492553613O5.7497158698N4.9536245179S0.1420323181)


### Mature leaves

Extract day-time starch:sucrose accumulation rate ratio from Lunn and Hatch 1995, Figure 1

<img src="references/Screenshot_20190714_104104.png" height=25% width=25%>

In [4]:
df_matureleaf_content_1 = pd.DataFrame(data={"Time":["Dawn","Dusk"],
                                             "Hexose (umol hexose/mgChl)":[24.126984126984112-22.222222222222204,
                                                                           121.26984126984127-116.19047619047619],
                                             "Sucrose (umol hexose/mgChl)":[22.222222222222204-17.142857142857146,
                                                                            116.19047619047619-95.87301587301586],
                                             "Starch (umol hexose/mgChl)":[17.142857142857146,95.87301587301586]})

In [5]:
Delta_sucrose = float(df_matureleaf_content_1["Sucrose (umol hexose/mgChl)"][df_matureleaf_content_1["Time"]=="Dusk"]) - \
float(df_matureleaf_content_1["Sucrose (umol hexose/mgChl)"][df_matureleaf_content_1["Time"]=="Dawn"])

Delta_hexose =float(df_matureleaf_content_1["Hexose (umol hexose/mgChl)"][df_matureleaf_content_1["Time"]=="Dusk"]) - \
float(df_matureleaf_content_1["Hexose (umol hexose/mgChl)"][df_matureleaf_content_1["Time"]=="Dawn"])

Delta_starch = float(df_matureleaf_content_1["Starch (umol hexose/mgChl)"][df_matureleaf_content_1["Time"]=="Dusk"]) - \
float(df_matureleaf_content_1["Starch (umol hexose/mgChl)"][df_matureleaf_content_1["Time"]=="Dawn"])

df_matureleaf_content_1["Starch (dusk-dawn)"]=["-",Delta_starch]
df_matureleaf_content_1["Sucrose (dusk-dawn)"]=["-",Delta_sucrose]
df_matureleaf_content_1["Hexose (dusk-dawn)"]=["-",Delta_hexose]
df_matureleaf_content_1

Hexose (umol hexose/mgChl)  Starch (umol hexose/mgChl)  \
0                    1.904762                   17.142857   
1                    5.079365                   95.873016   

   Sucrose (umol hexose/mgChl)  Time Starch (dusk-dawn) Sucrose (dusk-dawn)  \
0                     5.079365  Dawn                  -                   -   
1                    20.317460  Dusk            78.7302             15.2381   

  Hexose (dusk-dawn)  
0                  -  
1             3.1746

In [6]:
Sucrose_in_umols = 0.5*(df_matureleaf_content_1.at[1,"Sucrose (dusk-dawn)"])
Starch_in_umols = (df_matureleaf_content_1.at[1,"Starch (dusk-dawn)"])
Starch_to_sucrose_ratio = Starch_in_umols/Sucrose_in_umols
print ("Starch sucrose day-time accumulation rate = "+str(Starch_to_sucrose_ratio))

Starch sucrose day-time accumulation rate = 10.3333333333


### Leaf

Extract major sugar and amino acid content (8hr into photoperiod) from Ogbaga et al 2016, Figure 3, 4 and 5

<img src="references/Screenshot_20190715_102019.png" height=30% width=30%>

<img src="references/Screenshot_20190715_101103.png" height=50% width=50%>

<img src="references/Screenshot_20190715_102807.png" height=50% width=50%>

In [7]:
df_leaf_content_1 = pd.DataFrame(data={"Metabolite (mg/gDW)":["sucrose","glucose","fructose",
                                                              "ASP","SER",
                                                              "THR","ASN",
                                                              "PRO","PHE",
                                                              "TYR","TRP"],
                                       "variety Samsorg 17":[7.43944636678199,40.2479338842975,0.401372212692967,
                                                             0.0455882352941172,0.0260162601625953,
                                                             0.18373983739837,0.864705882352938,
                                                             0,0.00977272727272728,
                                                             0.0324175824175804,0],
                                       "variery Samsorg 40":[3.42560553633216,38.3471074380165,0,
                                                             0.0553921568627444,0.0357723577235695,
                                                             0.403252032520321,1.54705882352941,
                                                             0,0.0188636363636364,
                                                             0.0357142857142827,0]})
df_leaf_content_1.set_index("Metabolite (mg/gDW)",drop=True,inplace=True)
df_leaf_content_1

variery Samsorg 40  variety Samsorg 17
Metabolite (mg/gDW)                                        
sucrose                        3.425606            7.439446
glucose                       38.347107           40.247934
fructose                       0.000000            0.401372
ASP                            0.055392            0.045588
SER                            0.035772            0.026016
THR                            0.403252            0.183740
ASN                            1.547059            0.864706
PRO                            0.000000            0.000000
PHE                            0.018864            0.009773
TYR                            0.035714            0.032418
TRP                            0.000000            0.000000

Gathered soluble sugar, starch, cellulose, hemicellulose and lignin content from Wang et al 2013, Table 2

In [8]:
df_leaf_content_2 = pd.DataFrame(data={"Soluble sugar (mg/gDW)":[69.3],"Starch (mg/gDW)":[10.7],
                                       "Cellulose (mg/gDW)":[247.6],"Hemicellulose (mg/gDW)":[219.7],
                                       "Lignin (mg/gDW)":[13.2]})
df_leaf_content_2

Cellulose (mg/gDW)  Hemicellulose (mg/gDW)  Lignin (mg/gDW)  \
0               247.6                   219.7             13.2   

   Soluble sugar (mg/gDW)  Starch (mg/gDW)  
0                    69.3             10.7

Extracted phospholipid content from Vanhercke et al 2019 Figure 4

<img src="references/Screenshot_20190715_130438.png" height=100% width=100%>

In [9]:
df_leaf_content_3 = pd.DataFrame(data={"DAG (mmol/gDW)":[4.71830985915491],"DGDG (mmol/gDW)":[14.2253521126762],
                                       "LPC (mmol/gDW)":[0.985915492957744],"MGDG (mmol/gDW)":[13.8732394366197],
                                       "PC (mmol/gDW)":[4.71830985915491],"TAG (mmol/gDW)":[0.845070422535294]})
df_leaf_content_3

DAG (mmol/gDW)  DGDG (mmol/gDW)  LPC (mmol/gDW)  MGDG (mmol/gDW)  \
0         4.71831        14.225352        0.985915        13.873239   

   PC (mmol/gDW)  TAG (mmol/gDW)  
0        4.71831         0.84507

Estimate starch content using starch:sucrose ratio at dusk (from Lunn and Hatch 1995) and sucrose content (from Ogbaga et al 2016)

In [10]:
starch_mass = df_matureleaf_content_1["Starch (umol hexose/mgChl)"][1]* \
                                                    model.metabolites.get_by_id("GLC_c").formula_weight
sucrose_mass = df_matureleaf_content_1["Sucrose (umol hexose/mgChl)"][1]* \
                                                    model.metabolites.get_by_id("GLC_c").formula_weight

starch_mass2 = df_leaf_content_2["Starch (mg/gDW)"][0]
sucrose_mass2 = df_leaf_content_2["Soluble sugar (mg/gDW)"][0]
    

df_leaf_content_4 = pd.DataFrame(data={"Starch from Lunn and Hatch 1995 & Ogbaga et al 2016":
                                       [find_average(df_leaf_content_1.loc["sucrose"])*starch_mass/sucrose_mass],
                                       "Starch from Wang et al 2013 & Ogbaga et al 2016":
                                       [(find_average(df_leaf_content_1.loc["sucrose"])+
                                         find_average(df_leaf_content_1.loc["glucose"])+
                                         find_average(df_leaf_content_1.loc["fructose"]))*starch_mass2/sucrose_mass2]})
df_leaf_content_4

Starch from Lunn and Hatch 1995 & Ogbaga et al 2016  \
0                                          25.634732     

   Starch from Wang et al 2013 & Ogbaga et al 2016  
0                                         6.937357

### Summary

- Soluble sugar and amino acid data from Ogbaga et al 2016
- Phospholipid content was gathered from Vanhercke et al 2019

Assumptions

- Soluble sucrose content data from Ogbaga et al 2016 and Wang et al 2013 are very different. Values from the latter study was used to scale down, cellulose, hemicellulose and lignin
- Starch content predicted from Lunn and Hatch 1995 & Ogbaga et al 2016 is drastically different from that predicted from Starch from Wang et al 2013 & Ogbaga et al 2016. Values from the former was used for biomass.

In [11]:
biomass.at["sSUCROSE_b","leaf"]=find_average(df_leaf_content_1.loc["sucrose"])/ \
model.metabolites.get_by_id("SUCROSE_c").formula_weight

biomass.at["Starch_b","leaf"]=df_leaf_content_4["Starch from Lunn and Hatch 1995 & Ogbaga et al 2016"][0]/ \
                              model.metabolites.get_by_id("STARCH_p").formula_weight

biomass.at["GLC_c","leaf"]=find_average(df_leaf_content_1.loc["glucose"])/ \
model.metabolites.get_by_id("GLC_c").formula_weight
biomass.at["FRU_c","leaf"]=find_average(df_leaf_content_1.loc["fructose"])/ \
model.metabolites.get_by_id("FRU_c").formula_weight

starch_mass2 = df_leaf_content_2["Starch (mg/gDW)"][0]
sucrose_mass2 = df_leaf_content_2["Soluble sugar (mg/gDW)"][0]
biomass.at["Cellulose_b","leaf"]=df_leaf_content_2["Cellulose (mg/gDW)"]*(starch_mass2/sucrose_mass2)/ \
                                 model.metabolites.get_by_id("STARCH_p").formula_weight

biomass.at["DIACYLGLYCEROL_r","leaf"]=df_leaf_content_3["DAG (mmol/gDW)"][0]
biomass.at["PHOSPHATIDYL_CHOLINE_r","leaf"]=df_leaf_content_3["PC (mmol/gDW)"][0]

biomass.at["sASP_b","leaf"]=find_average(df_leaf_content_1.loc["ASP"])/ \
model.metabolites.get_by_id("L_ASPARTATE_c").formula_weight
biomass.at["sSER_b","leaf"]=find_average(df_leaf_content_1.loc["SER"])/ \
model.metabolites.get_by_id("SER_c").formula_weight
biomass.at["THR_c","leaf"]=find_average(df_leaf_content_1.loc["THR"])/ \
model.metabolites.get_by_id("THR_c").formula_weight
biomass.at["ASN_c","leaf"]=find_average(df_leaf_content_1.loc["ASN"])/ \
model.metabolites.get_by_id("ASN_c").formula_weight
biomass.at["PHE_c","leaf"]=find_average(df_leaf_content_1.loc["PHE"])/ \
model.metabolites.get_by_id("PHE_c").formula_weight
biomass.at["TYR_c","leaf"]=find_average(df_leaf_content_1.loc["TYR"])/ \
model.metabolites.get_by_id("TYR_c").formula_weight


### Stem



Extracted sucrose, glucose and fructose content from Njokweni 2015, Figure 3.1 and Figure 3.2

<img src="references/Screenshot_20190715_165703.png" height=50% width=50%>

<img src="references/Screenshot_20190715_170255.png" height=50% width=50%>

In [12]:
df_stem_content_1 = pd.DataFrame(data={"strain":["ICSB338","ICSB73","ICSV213","S35"],
                                       "Sucrose (g/L)":[10.6730769230769,17.7884615384615,
                                                        9.18269230769231,0.048076923076926],
                                       "Glucose (g/L)":[8.13186813186791,2.8571428571426,
                                                        12.2527472527471,0.494505494505204],
                                       "Fructose (g/L)":[63.22097378277,21.6479400749044,
                                                         89.0636704119839,0],
                                       "Starch (g/L)":[11.2911392405063,6.1873417721519,
                                                       5.36708860759494,4.91139240506329]})
df_stem_content_1.set_index("strain",drop=True,inplace=True)
df_stem_content_1

Fructose (g/L)  Glucose (g/L)  Starch (g/L)  Sucrose (g/L)
strain                                                             
ICSB338       63.220974       8.131868     11.291139      10.673077
ICSB73        21.647940       2.857143      6.187342      17.788462
ICSV213       89.063670      12.252747      5.367089       9.182692
S35            0.000000       0.494505      4.911392       0.048077

Extracted sucrose, starch, glucose, fructose, protein, arabinan, galactan, lignin, xylan and cellulose content of -6 DPA stem from McKinley et al 2016 Figure 3

<img src="references/Screenshot_20190715_170731.png" height=75% width=75%>

In [13]:
df_stem_content_2 = pd.DataFrame(data={"Metabolites":["sucrose","starch","glucose","fructose","protein",
                                                      "arabinan","galactan","lignin","xylan","cellulose"],
                                       "-6 DPA (g/100gDW)":[3.65614798694232,2.48787426500548,8.29162132752991,6.98585418933624,
                                                 1.63220892274211,1.17519042437434,0.457018498367745,11.8824809575626,
                                                 14.8204570184984,28.8030467899891]})
df_stem_content_2.set_index("Metabolites",drop=True,inplace=True)
df_stem_content_2["-6 DPA (mg/gDW)"]=df_stem_content_2["-6 DPA (g/100gDW)"]*10
df_stem_content_2

-6 DPA (g/100gDW)  -6 DPA (mg/gDW)
Metabolites                                    
sucrose               3.656148        36.561480
starch                2.487874        24.878743
glucose               8.291621        82.916213
fructose              6.985854        69.858542
protein               1.632209        16.322089
arabinan              1.175190        11.751904
galactan              0.457018         4.570185
lignin               11.882481       118.824810
xylan                14.820457       148.204570
cellulose            28.803047       288.030468

Retrieved average soluble sugar, starch, hemicellulose, cellulose and lignin content from Arai-Sanoh et al 2011, Table 2

In [14]:
df_stem_content_3 = pd.DataFrame(data={"soluble sugar (mg/gDW)":[256],"starch (mg/gDW)":[28],
                                       "hemicellulose (mg/gDW)":[164],"cellulose (mg/gDW)":[305],
                                       "lignin (mg/gDW)":[128]})
df_stem_content_3

cellulose (mg/gDW)  hemicellulose (mg/gDW)  lignin (mg/gDW)  \
0                 305                     164              128   

   soluble sugar (mg/gDW)  starch (mg/gDW)  
0                     256               28

Gathered soluble sugar, starch, cellulose, hemicellulose and lignin content from Wang et al 2013, Table 2

In [15]:
df_stem_content_4 = pd.DataFrame(data={"Soluble sugar (mg/gDW)":[157.0],"Starch (mg/gDW)":[9.8],
                                       "Cellulose (mg/gDW)":[281.8],"Hemicellulose (mg/gDW)":[251.3],
                                       "Lignin (mg/gDW)":[32.5]})
df_stem_content_4

Cellulose (mg/gDW)  Hemicellulose (mg/gDW)  Lignin (mg/gDW)  \
0               281.8                   251.3             32.5   

   Soluble sugar (mg/gDW)  Starch (mg/gDW)  
0                   157.0              9.8

Gathered sucrose, fructose, glucose, amino acids, organic acids, trehalose, maltose, mannitol, mannose, galactose and raffinose relative abundance from Njokweni 2015 table 2.1

In [16]:
df_stem_content_5 = pd.DataFrame(data={"strain":["ICSB338","ICSB73","ICSV213","S35"],
                                       "Ala":[9.02,2.55,3.7,1.93],
                                       "Val":[0.34,0,12.6,1.25],
                                       "Ser":[1.46,0.26,1.81,1.45],
                                       "Pro":[0.16,0.14,1.01,0.7],
                                       "Gln":[0.25,0,0.16,0.23],
                                       "Leu":[8.7,4.8,49.4,25],
                                       "Gly":[0.61,0.24,0.2,0.15],
                                       "Asp":[1.45,0.34,1.57,1.88],
                                       "Malate":[7.28,5.22,7.28,3.78],
                                       "Trehalose":[0,0.89,0.82,0.41],
                                       "maltose":[0,0.61,0,0],
                                       "Mannitol":[3.12,1.72,0.18,0.17],
                                       "Mannose":[0.31,4.02,0.27,1.12],
                                       "Galactose":[2.16,7.05,1.13,0.24],
                                       "Raffinose":[0,0.21,0,0],
                                       "Glucose":[2.6,8.63,1.35,2.17],
                                       "Fructose":[1.87,7.49,1.18,6.7],
                                       "Sucrose":[61.8,64.5,94.1,39.2]})
df_stem_content_5.set_index("strain",drop=True,inplace=True)
df_stem_content_5

Ala   Asp  Fructose  Galactose   Gln  Glucose   Gly   Leu  Malate  \
strain                                                                        
ICSB338  9.02  1.45      1.87       2.16  0.25     2.60  0.61   8.7    7.28   
ICSB73   2.55  0.34      7.49       7.05  0.00     8.63  0.24   4.8    5.22   
ICSV213  3.70  1.57      1.18       1.13  0.16     1.35  0.20  49.4    7.28   
S35      1.93  1.88      6.70       0.24  0.23     2.17  0.15  25.0    3.78   

         Mannitol  Mannose   Pro  Raffinose   Ser  Sucrose  Trehalose    Val  \
strain                                                                         
ICSB338      3.12     0.31  0.16       0.00  1.46     61.8       0.00   0.34   
ICSB73       1.72     4.02  0.14       0.21  0.26     64.5       0.89   0.00   
ICSV213      0.18     0.27  1.01       0.00  1.81     94.1       0.82  12.60   
S35          0.17     1.12  0.70       0.00  1.45     39.2       0.41   1.25   

         maltose  
strain            
ICSB338     0.00  
ICSB73      0.61  
ICSV213     0.00  
S35         0.00

In [17]:
biomass.at["sSUCROSE_b","stem"] =  df_stem_content_2.at["sucrose","-6 DPA (mg/gDW)"]/ \
                                       model.metabolites.get_by_id("SUCROSE_c").formula_weight
biomass.at["GLC_c","stem"]=df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
                                       model.metabolites.get_by_id("GLC_c").formula_weight
biomass.at["FRU_c","stem"]=df_stem_content_2.at["fructose","-6 DPA (mg/gDW)"]/ \
                                       model.metabolites.get_by_id("FRU_c").formula_weight
biomass.at["Cellulose_b","stem"]=df_stem_content_2.at["cellulose","-6 DPA (mg/gDW)"]/ \
                                       model.metabolites.get_by_id("STARCH_p").formula_weight

#scaling=find_average(df_stem_content_5["Malate"])/find_average(df_stem_content_5["Glucose"])
#biomass.at["sMAL_b","stem"] =scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("MAL_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Asp"])/find_average(df_stem_content_5["Glucose"])
#biomass.at["sASP_b","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("L_ASPARTATE_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Ser"])/find_average(df_stem_content_5["Glucose"])
#biomass.at["sSER_b","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("SER_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Gln"])/find_average(df_stem_content_5["Glucose"])
#biomass.at["sGLN_b","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("GLN_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Gly"])/find_average(df_stem_content_5["Glucose"])
#biomass.at["GLY_c","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("PRO_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Ala"])/find_average(df_stem_content_5["Glucose"])    
#biomass.at["sALA_b","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("L_ALPHA_ALANINE_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Val"])/find_average(df_stem_content_5["Glucose"])    
#biomass.at["VAL_c","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("VAL_c").formula_weight
#
#scaling=find_average(df_stem_content_5["Leu"])/find_average(df_stem_content_5["Glucose"])    
#biomass.at["LEU_c","stem"]=scaling*df_stem_content_2.at["glucose","-6 DPA (mg/gDW)"]/ \
#                                       model.metabolites.get_by_id("LEU_c").formula_weight

### Summary

- sucrose, glucose and fructose content from Njokweni 2015 are in g/L. Conversion to mg/gDW is not available and hence these data points are ignored

Assumptions

- Although r

### Root

Retrieve sugar, amino acid and organic acid content from Cambria et al 1983, Table 1

In [18]:
df_root_content_1 = pd.DataFrame(data={"Variety":["Al-sensitive","Al-tolerant"],
                                       "Sugars (mg/gFW)":[4.50,7.50],
                                       "Amino acids (umoles/gFW)":[7.40,7.10],
                                       "organic acids (ueq/gFW)":[16.3,23.64]})
df_root_content_1.set_index("Variety",drop=True,inplace=True)
df_root_content_1

Amino acids (umoles/gFW)  Sugars (mg/gFW)  \
Variety                                                   
Al-sensitive                       7.4              4.5   
Al-tolerant                        7.1              7.5   

              organic acids (ueq/gFW)  
Variety                                
Al-sensitive                    16.30  
Al-tolerant                     23.64

Retrieve organic acid content from Cambria et al 1983, Table 2

In [19]:
df_root_content_2 = pd.DataFrame(data={"Variety":["Al-sensitive","Al-tolerant"],
                                       "Succinic acid":[0.24,0.17],
                                       "Fumaric acid":[1.44,1.70],
                                       "Malic acid":[3.60,6.53],
                                       "t-Aconitic acid":[2.40,9.50],
                                       "Quinic acid":[0.04,0.17],
                                       "Citric acid":[0.50,0.61],
                                       "Total":[8.22,18.68]})
df_root_content_2.set_index("Variety",drop=True,inplace=True)
df_root_content_2

Citric acid  Fumaric acid  Malic acid  Quinic acid  \
Variety                                                            
Al-sensitive         0.50          1.44        3.60         0.04   
Al-tolerant          0.61          1.70        6.53         0.17   

              Succinic acid  Total  t-Aconitic acid  
Variety                                              
Al-sensitive           0.24   8.22              2.4  
Al-tolerant            0.17  18.68              9.5

### Seed


Retrieve soluble sugar, starch, cellulose, hemicellulose and lignin content from Wang et al 2013 Table 2

In [20]:
df_seed_content_1 = pd.DataFrame(data={"Soluble sugar (mg/gDW)":[52.3],"Starch (mg/gDW)":[395.2],
                                     "Cellulose (mg/gDW)":[43.7],"Hemicellulose (mg/gDW)":[147.2],
                                     "Lignin (mg/gDW)":[8.1]})
df_seed_content_1

Cellulose (mg/gDW)  Hemicellulose (mg/gDW)  Lignin (mg/gDW)  \
0                43.7                   147.2              8.1   

   Soluble sugar (mg/gDW)  Starch (mg/gDW)  
0                    52.3            395.2

Estimated FW/DW ratio from Neucere et al 1980, Table I

In [21]:
Moisture = 12.07
DW = 100 - Moisture
FW_DW_ratio = 100/DW
print FW_DW_ratio


1.13726828159


Retrieved fructose, glucose, sucrose, maltose and raffinose content from Neucere et al 1980, Table I and IV

In [22]:
df_seed_content_2 = pd.DataFrame(data={"variety":["SC301","NSA740","CK60","TX615","GA615"],
                                       "fructose (g/100gDW)":[1.67,1.12,2.32,1.80,1.06],
                                       "glucose (g/100gDW)":[1.78,0.68,2,2.94,0.87],
                                       "sucrose (g/100gDW)":[0.51,0.30,0.58,0.38,0.30],
                                       "maltose (g/100gDW)":[0.35,0.16,0.23,0.78,0.06],
                                       "raffinose (g/100gDW)":[0.1,0.03,0.03,0.11,0.05],
                                       "protein (g/100gFW)":[11.48,14.32,9.90,11.85,9.75],
                                       "fat (g/100gFW)":[3.03,3.49,3.31,2.66,3.00]})
df_seed_content_2_std = pd.DataFrame(data={"variety":["SC301","NSA740","CK60","TX615","GA615"],
                                           "fructose (mg/gDW)":df_seed_content_2["fructose (g/100gDW)"]*10,
                                           "glucose (mg/gDW)":df_seed_content_2["glucose (g/100gDW)"]*10,
                                           "sucrose (mg/gDW)":df_seed_content_2["sucrose (g/100gDW)"]*10,
                                           "maltose (mg/gDW)":df_seed_content_2["maltose (g/100gDW)"]*10,
                                           "raffinose (mg/gDW)":df_seed_content_2["raffinose (g/100gDW)"]*10,
                                           "protein (mg/gDW)":(df_seed_content_2["protein (g/100gFW)"]*FW_DW_ratio)*10,
                                           "fat (mg/gDW)":(df_seed_content_2["fat (g/100gFW)"]*FW_DW_ratio)*10})

df_seed_content_2_std.set_index("variety",drop=True,inplace=True)
df_seed_content_2_std

fat (mg/gDW)  fructose (mg/gDW)  glucose (mg/gDW)  maltose (mg/gDW)  \
variety                                                                        
SC301        26.64279               16.7              17.8               3.5   
NSA740       30.68757               11.2               6.8               1.6   
CK60         29.10483               23.2              20.0               2.3   
TX615        23.38938               18.0              29.4               7.8   
GA615        26.37900               10.6               8.7               0.6   

         protein (mg/gDW)  raffinose (mg/gDW)  sucrose (mg/gDW)  
variety                                                          
SC301           100.94364                 1.0               5.1  
NSA740          125.91576                 0.3               3.0  
CK60             87.05070                 0.3               5.8  
TX615           104.19705                 1.1               3.8  
GA615            85.73175                 0.5               3.0

Gather amino acid composition from Mosse, Huet and Baudet 1988

In [23]:
df_seed_content_3 = pd.DataFrame(data={"variety":range(1,13),
                                       "N content (g/100gDW)":[1.51,1.67,1.74,1.86,1.86,1.88,
                                                               2.06,2.25,2.34,2.66,2.86,2.97],
                                       "Gly (g/16g N)":[3.4,3.15,3.05,3.05,3.1,2.9,2.85,2.8,3,2.75,2.55,2.5],
                                       "Ala (g/16g N)":[9.1,9.2,9.2,9.4,9.5,9.1,9.6,9.5,9.3,9.8,9.7,10],
                                       "Val (g/16g N)":[5.2,5.5,5.3,5.6,5.6,5.2,5.6,5.3,5.3,5.3,5.2,5.1],
                                       "Leu (g/16g N)":[12.7,13.9,13.6,13.8,14.2,13.6,
                                                        14.4,13.8,13.3,14.2,14.1,14.7],
                                       "Ile (g/16g N)":[4.1,4.3,4.2,4.3,4.4,4.1,4.3,4.2,4.2,4.4,4.3,4.4],
                                       "Ser (g/16g N)":[4.7,4.6,4.5,4.6,4.7,4.5,4.5,4.7,4.6,4.7,4.5,4.5],
                                       "Thr (g/16g N)":[3.35,3.25,3.15,3.3,3.3,3.2,3.15,3.2,3.2,3.2,3.1,3],
                                       "Tyr (g/16g N)":[3.9,4.2,4.1,4.2,4.3,3.9,4.3,3.9,4,4,4,4.1],
                                       "Phe (g/16g N)":[5.2,5.5,5.4,5.5,5.5,5.2,5.4,5.5,5.3,5.5,5.6,5.7],
                                       "Trp (g/16g N)":[1.08,1.03,1,1.01,1.11,0.84,1,1.05,1.05,1.02,1,1],
                                       "Pro (g/16g N)":[8.1,8.8,8.5,8.6,8.9,8.2,8.9,8.3,8.1,8.4,8.3,8.4],
                                       "Met (g/16g N)":[2,1.65,1.55,1.6,1.7,1.35,1.55,1.35,1.65,1.6,1.4,1.4],
                                       "Half Cys (g/16g N)":[2.15,1.9,1.7,1.75,1.9,1.75,
                                                             1.65,1.85,1.9,1.6,1.45,1.5],
                                       "Lys (g/16g N)":[2.5,2.3,2.3,2.3,2.3,2.1,2.15,2.05,2.3,1.95,1.95,1.65],
                                       "His (g/16g N)":[2.3,2.2,2.15,2.25,2.3,2.2,2.15,2.3,2.1,2.05,2,2],
                                       "Arg (g/16g N)":[4.1,4,4,4,4.1,3.8,3.8,3.7,4.2,3.7,3.7,3.1],
                                       "Asp (g/100gDW)":[6.7,7,7,7,6.9,9.7,7,7.7,7.3,7.3,7.1,6.7],
                                       "Glt (g/100gDW)":[20.3,21,20.8,21.4,21.7,20.8,
                                                         22,22.4,21.2,22.6,22.3,22.9]})
df_seed_content_3_std = pd.DataFrame(data={"variety":range(1,13),
                                       "Gly (mg/gDW)":(df_seed_content_3["Gly (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Ala (mg/gDW)":(df_seed_content_3["Ala (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Val (mg/gDW)":(df_seed_content_3["Val (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Leu (mg/gDW)":(df_seed_content_3["Leu (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Ile (mg/gDW)":(df_seed_content_3["Ile (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Ser (mg/gDW)":(df_seed_content_3["Ser (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Thr (mg/gDW)":(df_seed_content_3["Thr (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Tyr (mg/gDW)":(df_seed_content_3["Tyr (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Phe (mg/gDW)":(df_seed_content_3["Phe (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Trp (mg/gDW)":(df_seed_content_3["Trp (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Pro (mg/gDW)":(df_seed_content_3["Pro (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Met (mg/gDW)":(df_seed_content_3["Met (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Cys (mg/gDW)":0.5*(df_seed_content_3["Half Cys (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Lys (mg/gDW)":(df_seed_content_3["Lys (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "His (mg/gDW)":(df_seed_content_3["His (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Arg (mg/gDW)":(df_seed_content_3["Arg (g/16g N)"]/16)* \
                                                            df_seed_content_3["N content (g/100gDW)"]*10,
                                       "Asp (mg/gDW)":df_seed_content_3["Asp (g/100gDW)"]*10,
                                       "Glt (mg/gDW)":df_seed_content_3["Glt (g/100gDW)"]*10})
df_seed_content_3_std.set_index("variety",drop=True,inplace=True)
df_seed_content_3_std

Ala (mg/gDW)  Arg (mg/gDW)  Asp (mg/gDW)  Cys (mg/gDW)  Glt (mg/gDW)  \
variety                                                                         
1            8.588125      3.869375          67.0      1.014531         203.0   
2            9.602500      4.175000          70.0      0.991562         210.0   
3           10.005000      4.350000          70.0      0.924375         208.0   
4           10.927500      4.650000          70.0      1.017188         214.0   
5           11.043750      4.766250          69.0      1.104375         217.0   
6           10.692500      4.465000          97.0      1.028125         208.0   
7           12.360000      4.892500          70.0      1.062188         220.0   
8           13.359375      5.203125          77.0      1.300781         224.0   
9           13.601250      6.142500          73.0      1.389375         212.0   
10          16.292500      6.151250          73.0      1.330000         226.0   
11          17.338750      6.613750          71.0      1.295937         223.0   
12          18.562500      5.754375          67.0      1.392187         229.0   

         Gly (mg/gDW)  His (mg/gDW)  Ile (mg/gDW)  Leu (mg/gDW)  Lys (mg/gDW)  \
variety                                                                         
1            3.208750      2.170625      3.869375     11.985625      2.359375   
2            3.287812      2.296250      4.488125     14.508125      2.400625   
3            3.316875      2.338125      4.567500     14.790000      2.501250   
4            3.545625      2.615625      4.998750     16.042500      2.673750   
5            3.603750      2.673750      5.115000     16.507500      2.673750   
6            3.407500      2.585000      4.817500     15.980000      2.467500   
7            3.669375      2.768125      5.536250     18.540000      2.768125   
8            3.937500      3.234375      5.906250     19.406250      2.882812   
9            4.387500      3.071250      6.142500     19.451250      3.363750   
10           4.571875      3.408125      7.315000     23.607500      3.241875   
11           4.558125      3.575000      7.686250     25.203750      3.485625   
12           4.640625      3.712500      8.167500     27.286875      3.062813   

         Met (mg/gDW)  Phe (mg/gDW)  Pro (mg/gDW)  Ser (mg/gDW)  Thr (mg/gDW)  \
variety                                                                         
1            1.887500      4.907500      7.644375      4.435625      3.161563   
2            1.722187      5.740625      9.185000      4.801250      3.392188   
3            1.685625      5.872500      9.243750      4.893750      3.425625   
4            1.860000      6.393750      9.997500      5.347500      3.836250   
5            1.976250      6.393750     10.346250      5.463750      3.836250   
6            1.586250      6.110000      9.635000      5.287500      3.760000   
7            1.995625      6.952500     11.458750      5.793750      4.055625   
8            1.898438      7.734375     11.671875      6.609375      4.500000   
9            2.413125      7.751250     11.846250      6.727500      4.680000   
10           2.660000      9.143750     13.965000      7.813750      5.320000   
11           2.502500     10.010000     14.836250      8.043750      5.541250   
12           2.598750     10.580625     15.592500      8.353125      5.568750   

         Trp (mg/gDW)  Tyr (mg/gDW)  Val (mg/gDW)  
variety                                            
1            1.019250      3.680625      4.907500  
2            1.075063      4.383750      5.740625  
3            1.087500      4.458750      5.763750  
4            1.174125      4.882500      6.510000  
5            1.290375      4.998750      6.510000  
6            0.987000      4.582500      6.110000  
7            1.287500      5.536250      7.210000  
8            1.476563      5.484375      7.453125  
9            1.535625      5.850000      7.751250  
10           1.695750      6.650

In [24]:
biomass.at["sSUCROSE_b","seed"] = find_average(df_seed_content_2_std["sucrose (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("SUCROSE_c").formula_weight
biomass.at["GLC_c","seed"]=find_average(df_seed_content_2_std["glucose (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("GLC_c").formula_weight
biomass.at["FRU_c","seed"]=find_average(df_seed_content_2_std["fructose (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("FRU_c").formula_weight
biomass.at["L_PHOSPHATIDATE_p","leaf"]=14.2253521126762/model.metabolites.get_by_id("L_PHOSPHATIDATE_p").formula_weight
biomass.at["Protein_b","leaf"]=8.339/proteinMW

biomass.at["ARG_c","seed"] = find_average(df_seed_content_3_std["Arg (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("ARG_c").formula_weight
biomass.at["HIS_c","seed"] = find_average(df_seed_content_3_std["His (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("HIS_c").formula_weight
biomass.at["LYS_c","seed"] = find_average(df_seed_content_3_std["Lys (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("LYS_c").formula_weight
biomass.at["sASP_b","seed"] = find_average(df_seed_content_3_std["Asp (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("L_ASPARTATE_c").formula_weight
biomass.at["sGLU_b","seed"] = find_average(df_seed_content_3_std["Glt (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("sGLU_b").formula_weight
biomass.at["sSER_b","seed"] = find_average(df_seed_content_3_std["Ser (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("sSER_b").formula_weight
biomass.at["THR_c","seed"] = find_average(df_seed_content_3_std["Thr (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("THR_c").formula_weight
biomass.at["CYS_c","seed"] = find_average(df_seed_content_3_std["Cys (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("CYS_c").formula_weight
biomass.at["GLY_c","seed"] = find_average(df_seed_content_3_std["Gly (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("GLY_c").formula_weight
biomass.at["PRO_c","seed"] = find_average(df_seed_content_3_std["Pro (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("PRO_c").formula_weight
biomass.at["sALA_b","seed"] = find_average(df_seed_content_3_std["Ala (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("L_ALPHA_ALANINE_c").formula_weight
biomass.at["VAL_c","seed"] = find_average(df_seed_content_3_std["Val (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("VAL_c").formula_weight
biomass.at["ILE_c","seed"] = find_average(df_seed_content_3_std["Ile (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("ILE_c").formula_weight
biomass.at["LEU_c","seed"] = find_average(df_seed_content_3_std["Leu (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("LEU_c").formula_weight
biomass.at["MET_c","seed"] = find_average(df_seed_content_3_std["Met (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("MET_c").formula_weight
biomass.at["PHE_c","seed"] = find_average(df_seed_content_3_std["Phe (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("PHE_c").formula_weight
biomass.at["TYR_c","seed"] = find_average(df_seed_content_3_std["Tyr (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("TYR_c").formula_weight
biomass.at["TRP_c","seed"] = find_average(df_seed_content_3_std["Trp (mg/gDW)"])/ \
                                            model.metabolites.get_by_id("TRP_c").formula_weight
biomass.to_csv("/home/sanu/biomass_sorghum.csv")

### References

- Wang, J.S., Wang, M.L., Spiertz, J.H.J., Liu, Z.X., Han, L.P. and Xie, G.H. (2013) Genetic variation in yield and chemical composition of wide range of sorghum accessions grown in north-west China. Res. Crop., 14, 95–105.
- Ogbaga, C.C., Stepien, P., Dyson, B.C., Rattray, N.J.W., Ellis, D.I., Goodacre, R. and Johnson, G.N. (2016) Biochemical analyses of sorghum varieties reveal differential responses to drought J.-S. Zhang, ed. PLoS One, 11, e0154423. https://doi.org/10.1371/journal.pone.0154423.
- Vanhercke, T., Belide, S., Taylor, M.C., et al. (2019) Up-regulation of lipid biosynthesis increases the oil content in leaves of Sorghum bicolor. Plant Biotechnol. J., 17, 220–232. Available at: http://doi.wiley.com/10.1111/pbi.12959.
- Lunn, J.E. and Hatch, M.D. (1995) Primary partitioning and storage of photosynthate in sucrose and starch in leaves of C4 plants. Planta, 197, 5000. https://doi.org/10.1007/BF00202661.
- Njokweni, A.P. (University of the W.C. (2015) COMPARATIVE ANALYSIS OF SUGAR-BIOSYNTHESIS PROTEINS OF SORGHUM STEMS AND THE INVESTIGATION OF THEIR ROLE IN HYPEROSMOTIC STRESS TOLERANCE. University of the Western Cape. 
https://pdfs.semanticscholar.org/cb3f/6b1c4f9b2acae9fe0afba1c3603c32fdb2c7.pdf.
- Arai-Sanoh, Y., Ida, M., Zhao, R., et al. (2011) Genotypic variations in non-structural carbohydrate and cell-wall components of the stem in rice, sorghum, and sugar vane. Biosci. Biotechnol. Biochem., 75, 1104–12. Available at: http://www.tandfonline.com/doi/full/10.1271/bbb.110009.
- Cambria, J., Galvani, F.R., Estevao, M.M. and Sant’Anna, R. (1983) Effects of Aluminum on Organic Acid,’ Sugar and Amino Acid Composition of the Root System of Sorghum (Sorghum Bicolor L. Moench). J. Plant Nutr., 6, 313–322. https://doi.org/10.1080/01904168309363092.
- Neucere, N.J. and Sumrell, G. (1980) Chemical composition of different varieties of grain sorghum. J. Agric. Food Chem., 28, 19–21. http://doi.org/10.1021/jf60227a022.